common/src/View/SetBrushFaceAttributesTool.cpp

```diff
@@ -47,26 +47,36 @@ namespace TrenchBroom {
         }
 
         bool SetBrushFaceAttributesTool::doMouseClick(const InputState& inputState) {
-            return performCopy(inputState, false);
+            if (canPerformCopy(inputState)) {
+                performCopy(inputState, false);
+                return true;
+            } else {
+                return false;
+            }
         }
 
         bool SetBrushFaceAttributesTool::doMouseDoubleClick(const InputState& inputState) {
-            return performCopy(inputState, true);
-        }
-
-        bool SetBrushFaceAttributesTool::performCopy(const InputState& inputState, const bool applyToBrush) {
-            if (!applies(inputState))
+            if (canPerformCopy(inputState)) {
+                // A double click is always preceeded by a single click, so we already done some work which is now
+                // superseded by what is done next. To avoid inconsistencies with undo, we undo the work done by the
+                // single click now:
+                auto document = kdl::mem_lock(m_document);
+                document->undoCommand();
+
+                performCopy(inputState, true);
+                return true;
+            } else {
                 return false;
+            }
+        }
 
+        void SetBrushFaceAttributesTool::performCopy(const InputState& inputState, const bool applyToBrush) {
```

Comment:
"Copy" is a confusing name here because it's more like pasting from the current selection. Actually I'd rather call it "transferFromSelection" maybe? "transferAttributesFromSelection"?

Reply:
I prefer the term copy because that's what it does (the attributes of the selected face stay intact), but I have renamed it to `copyAttributesFromSelection` because that's more descriptive.